File: get_before_data.py
import logging

from redis_connection import redis_client
from get_busuanzi_data import get_busuanzi_page_pv_data, get_busuanzi_site_uv_data, get_busuanzi_site_pv_data

def get_page_pv_before(host, path):
    """
    Retrieves the page view (PV) count for a specific page.

    Args:
    host (str): The host name.
    path (str): The path of the page.

    Returns:
    int: The page view count for the page.
    """
    page_key = f"live_page_pv:{host}{path}"
    page_pv = redis_client.get(page_key)
    logging.debug(f"page_pv: {page_pv}, page_key: {page_key}")
    if page_pv is None:
        page_pv = get_busuanzi_page_pv_data(host, path)["page_pv"]
        return page_pv
    else:
        return int(page_pv.decode())


def get_site_pv_before(host, path):
    """
    Retrieves the site page view (PV) count for a specific site.

    Args:
    host (str): The host name.
    path (str): The path of the page.

    Returns:
    int: The site page view count for the site.
    """
    site_key = f"live_site_pv:{host}"
    site_pv = redis_client.get(site_key)
    if site_pv is None:
        site_pv = get_busuanzi_site_pv_data(host, path)["site_pv"]
        return site_pv
    else:
        return int(site_pv.decode())


def get_site_uv_before(host, path):
    """
    Retrieves the site unique visitor (UV) count for a specific site.

    Args:
    host (str): The host name.
    path (str): The path of the page.

    Returns:
    int: The site unique visitor count for the site.
    """
    site_key = f"live_site_uv:{host}"
    site_uv = redis_client.get(site_key)
    if site_uv is None:
        site_uv = get_busuanzi_site_uv_data(host, path)["site_uv"]
        return site_uv
    else:
        return int(site_uv.decode())

File: src/lib/get-before-data.ts
import kv from "@/lib/kv";
import logger from "@/lib/logger";
import {
	getBusuanziPagePVData,
	getBusuanziSitePVData,
	getBusuanziSiteUVData,
} from "@/utils/busuanzi";

async function getSiteUVBeforeData(host: string, path: string) {
	const siteKey = `site_uv_live:${host}`;
	const siteUV = await kv.get(siteKey);
	if (!siteUV) {
		logger.debug(`site_uv not found for host: https://${host}${path}`);
		const siteUVData = await getBusuanziSiteUVData(host, path);
		const siteUV = siteUVData ? siteUVData : 0;
		// console.log(`site_uv_data: ${siteUVData}, site_uv: ${siteUV}`);
		return Number(siteUV);
	} else {
		// console.log(`site_uv: ${siteUV}`);
		logger.debug(
			`site_uv found for host: https://${host}${path}, site_uv: ${siteUV}`
		);
		return Number(siteUV);
	}
}

async function getSitePVBeforeData(host: string, path: string) {
	const siteKey = `site_pv_live:${host}`;
	const sitePV = await kv.get(siteKey);
	if (!sitePV) {
		logger.debug(`site_pv not found for host: https://${host}${path}`);
		const sitePVData = await getBusuanziSitePVData(host);
		const sitePV = sitePVData ? sitePVData : 0;
		logger.debug(`site_pv_data: ${sitePVData}, site_pv: ${sitePV}`);
		return Number(sitePV);
	} else {
		logger.debug(
			`site_pv found for host: https://${host}${path}, site_pv: ${sitePV}`
		);
		return Number(sitePV);
	}
}

async function getPagePVBeforeData(host: string, path: string) {
	const pageKey = `live_page_pv:${host}${path}`;
	const pagePV = await kv.get(pageKey);
	logger.debug(`page_pv: ${pagePV}, page_key: ${pageKey}`);

	if (!pagePV) {
		logger.debug(`page_pv not found for host: https://${host}${path}`);
		const pagePVData = await getBusuanziPagePVData(host, path);
		const pagePV = pagePVData ? pagePVData : 0;
		// console.log(`page_pv_data: ${pagePVData}, page_pv: ${pagePV}`);
		return Number(pagePV);
	} else {
		// console.log(`page_pv: ${pagePV}`);
		logger.debug(
			`page_pv found for host: https://${host}${path}, page_pv: ${pagePV}`
		);
		return Number(pagePV);
	}
}

export { getSiteUVBeforeData, getSitePVBeforeData, getPagePVBeforeData };
